### PR TITLE
Add support for vpc network configuration in google_service_directory_endpoint

### PR DESCRIPTION
**Pull request by lcaggio**
Add support for 'network' flag on ServiceDirectory Endpoint resource.

fixes https://github.com/hashicorp/terraform-provider-google/issues/10057

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
serviceDirectory: add support for vpc network configuration in google_service_directory_endpoint.
```

### CODE REVIEW

File: mmv1/products/servicedirectory/api.yaml [hunk 1]
```diff
@@ -178,4 +178,8 @@ objects:
           by service clients. The entire metadata dictionary may contain
           up to 512 characters, spread across all key-value pairs.
           Metadata that goes beyond any these limits will be rejected.
+      - !ruby/object:Api::Type::String
+        name: 'network'
+        description: |
+          The URL to the network, such as projects/PROJECT_NUMBER/locations/global/networks/NETWORK_NAME.
```
[the remark this answers is not in the document]
Comment (lcaggio) [reply mmv1/products/servicedirectory/api.yaml:184]: Unfortunately the project ID is not supported here, tested but it doesn't work.

The documentation mention that the PROJECT NUMBER is needed: https://cloud.google.com/sdk/gcloud/reference/beta/service-directory/endpoints/create

[the remark this answers is not in the document]
Comment (slevenick) [reply mmv1/products/servicedirectory/api.yaml:184]: That's unfortunate, but nothing to do from the provider side!

**Review by slevenick — APPROVED** [review]
Thanks!

**Pull request merged**